scenario/mocking.py

```diff
@@ -631,6 +632,13 @@ def resource_get(self, resource_name: str) -> str:
                 f"resource {resource_name} not found in State. please pass it.",
             )
 
+    def credential_get(self) -> CloudSpec:
+        if not self._state.cloud_spec:
+            raise ModelError(
```

Comment:
Side thought: I'm not sure this should be a ModelError, as it gives the impression that the error was raised by ops and not scenario. 
@tonyandrewmeyer what is the pattern in ops.testing? 
And can the hook tool error out and trigger a modelerror in production or is this a situation that can only occur in testing? 

If the latter, perhaps we should consider having a defaulted cloudspec

Reply:
> @tonyandrewmeyer what is the pattern in ops.testing?

`ModelError` if it's what would happen in prod (ie. we're pretending it's an error coming from Juju) and if it's something specific to Harness (the consistency-check type things and so on) then something else, often `RuntimeError`.

> And can the hook tool error out and trigger a modelerror in production or is this a situation that can only occur in testing?

It can - for example, it will always do that in K8s:

```
root@provider-charm-0:/var/lib/juju/agents/unit-provider-charm-0/charm# ../../../tools/unit-provider-charm-0/credential-get 
ERROR cannot access cloud credentials: credential-get on a "caas" model not supported
```

(ops will make that a `ModelError`). It will also end up as a `ModelError` if you haven't done `juju trust <app>`.

> If the latter, perhaps we should consider having a defaulted cloudspec

I think it's worthwhile having appropriate defaults for `scenario.CloudSpec` to make it easier to use it. I don't think having a default one is good, because then it's assuming lxd and trust (we could get the model type but not when creating a `State` since that's in the context, and I don't think Scenario currently has any modelling of (Juju) trust).